tests/client/handlers/vaults.test.ts

```diff
@@ -1470,6 +1477,90 @@ describe('vaultsSecretsNew and vaultsSecretsDelete, vaultsSecretsGet', () => {
+    const secret1 = 'test-secret1';
+    const secret2 = 'test-secret2';
+    const vaultId = await vaultManager.createVault('test-vault');
+    const vaultIdEncoded = vaultsUtils.encodeVaultId(vaultId);
+    await rpcClient.methods.vaultsSecretsNew({
+      nameOrId: vaultIdEncoded,
+      secretName: secret1,
+      secretContent: Buffer.from(secret1).toString('binary'),
+    });
+    await rpcClient.methods.vaultsSecretsNew({
+      nameOrId: vaultIdEncoded,
+      secretName: secret2,
+      secretContent: Buffer.from(secret2).toString('binary'),
+    });
+    // Get secret
+    const getResponse1 = await rpcClient.methods.vaultsSecretsGet({
+      nameOrId: vaultIdEncoded,
+      secretName: secret1,
+    });
+    const getResponse2 = await rpcClient.methods.vaultsSecretsGet({
+      nameOrId: vaultIdEncoded,
+      secretName: secret2,
+    });
+    expect(getResponse1.secretContent).toStrictEqual(secret1);
+    expect(getResponse2.secretContent).toStrictEqual(secret2);
+    // Delete secret
+    const deleteResponse = await rpcClient.methods.vaultsSecretsRemove({
+      nameOrId: vaultIdEncoded,
+      secretNames: [secret1, secret2],
+    });
+    expect(deleteResponse.success).toBeTruthy();
+    // Check secret was deleted
+    await testsUtils.expectRemoteError(
+      rpcClient.methods.vaultsSecretsGet({
+        nameOrId: vaultIdEncoded,
+        secretName: secret1,
+      }),
+      vaultsErrors.ErrorSecretsSecretUndefined,
+    );
+    await testsUtils.expectRemoteError(
+      rpcClient.methods.vaultsSecretsGet({
+        nameOrId: vaultIdEncoded,
+        secretName: secret2,
+      }),
+      vaultsErrors.ErrorSecretsSecretUndefined,
+    );
+  });
+  test('deletes directory recursively', async () => {
+    // Create secret
+    const vaultName = 'test-vault';
+    const secretList = ['test-secret1', 'test-secret2', 'test-secret3'];
+    const secretDir = path.join(dataDir, 'secretDir');
+    await fs.promises.mkdir(secretDir);
+    for (const secret of secretList) {
+      const secretFile = path.join(secretDir, secret);
+      // Write secret to file
+      await fs.promises.writeFile(secretFile, secret);
+    }
+    const vaultId = await vaultManager.createVault(vaultName);
+    const vaultsIdEncoded = vaultsUtils.encodeVaultId(vaultId);
+    const addResponse = await rpcClient.methods.vaultsSecretsNewDir({
+      nameOrId: vaultsIdEncoded,
+      dirName: secretDir,
+    });
+    expect(addResponse.success).toBeTruthy();
+    // Delete secret
+    await testsUtils.expectRemoteError(
+      rpcClient.methods.vaultsSecretsRemove({
+        nameOrId: vaultsIdEncoded,
+        secretNames: ['secretDir'],
+      }),
+      vaultsErrors.ErrorVaultsRecursive,
+    );
+    const deleteResponse = await rpcClient.methods.vaultsSecretsRemove({
+      nameOrId: vaultsIdEncoded,
+      secretNames: ['secretDir'],
+      options: {
+        recursive: true,
+      },
+    });
+    expect(deleteResponse.success).toBeTruthy();
```

Comment:
You still need to check that each secret is removed as expected including the directory.